### DISCUSSION TITLE
Images from media library not showing after migrate to another server

### DISCUSSION THREAD

**Discussion by Tauxxx** [area17/twill#2679] [General]
I migrated the project from a local server to a remote server. In the admin panel, images uploaded through the media library are not displaying, and when trying to open the image file in the browser, I get a 404 error. Even if I upload a new image, it appears in the file system, but the image itself doesn't display in the browser (the same 404 error).

Of course, I have run php artisan storage:link and checked file permissions. The path to the image looks like this: /img/75efb280-1c52-4781-9afc-6f9690d54452/1s.jpg.

**Comment by kallefrombosnia** [comment]
If you open the image manualy in the browser, does it show up?
Usualy, if debugging is enabled, you will get a message about the issue.

**Reply by Tauxxx** [reply]
> If you open the image manualy in the browser, does it show up? Usualy, if debugging is enabled, you will get a message about the issue.

No, I only see 404.

**Comment by Tauxxx** [comment]
To rule out any errors during the migration, I completely removed everything and reinstalled Twill CMS from scratch. However, I encountered the exact same issue with a 404 error. The image uploads successfully but does not display in the browser.

**Comment by ifox** [comment]
Multiple things could be causing this. Since you do not have any error message, the first thing that comes to mind is that your webserver may be configured to server static files under the /img path. If that's the case, you either need to remove that and let the /img path go through your Laravel application, or change the path using `GLIDE_BASE_PATH`. I know that sounds unlikely, but it is a possibility. 

Next, I'd check if the PHP GD extension is enabled.

If you're still not successful, please share your full media library configuration so I can advise more. You can output `config("twill.media_library")` through Tinker to make sure to share the active configuration.

**Reply by Tauxxx** [reply]
> Multiple things could be causing this. Since you do not have any error message, the first thing that comes to mind is that your webserver may be configured to server static files under the /img path. If that's the case, you either need to remove that and let the /img path go through your Laravel application, or change the path using `GLIDE_BASE_PATH`. I know that sounds unlikely, but it is a possibility.
> 
> Next, I'd check if the PHP GD extension is enabled.
> 
> If you're still not successful, please share your full media library configuration so I can advise more. You can output `config("twill.media_library")` through Tinker to make sure to share the active configuration.

1/ There are no rules for "img" in the Nginx configuration.  
2/ I checked the PHP GD configuration - it’s enabled. I also tested the `imagecopyresampled` function, and the image cropped without any issues.  
3/ Configuration is provided below.

In fact, this setup follows the standard installation instructions without any configuration modifications.

````
 config('twill.media_library');
= [
    "disk" => "twill_media_library",
    "endpoint_type" => "local",
    "cascade_delete" => false,
    "local_path" => "uploads",
    "image_service" => "A17\Twill\Services\MediaLibrary\Glide",
    "acl" => "private",
    "filesize_limit" => 50,
    "allowed_extensions" => [
      "svg",
      "jpg",
      "gif",
      "png",
      "jpeg",
    ],
    "init_alt_text_from_filename" => true,
    "prefix_uuid_with_local_path" => false,
    "translated_form_fields" => false,
    "show_file_name" => false,
    "media_caption_use_wysiwyg" => false,
    "media_caption_wysiwyg_options" => [
      "modules" => [
        "toolbar" => [
          "bold",
          "italic",
        ],
      ],
    ],
  ]
```

**Comment by Tauxxx** [comment]
As it turned out, it was all about nginx. My starndart config nginx inculdes rules for caching static files:
```
location ~* \.(jpg|jpeg|gif|png|css|zip|tgz|gz|..... {
 access_log off;
 expires max;
 log_not_found off;
}
```
and this rule defines how image files will be processed. Just added in this rule:
```try_files $uri $uri/ /index.php?$query_string;```
